phoenix-core/src/main/java/org/apache/phoenix/hbase/index/Indexer.java

```diff
@@ -800,5 +804,62 @@ public static void enableIndexing(HTableDescriptor desc, Class<? extends IndexBu
     properties.put(Indexer.INDEX_BUILDER_CONF_KEY, builder.getName());
     desc.addCoprocessor(Indexer.class.getName(), null, priority, properties);
   }
+
+    /**
+     * stop indexer when region split, but split failed, and then rollback, open region.
+     * this need init builder and writer,recoveryWriter.
+     *
+     * @param env
+     */
+  private void initBuilderAndWriter(RegionCoprocessorEnvironment env) {
+      // init index build manager
+      try {
+          this.builder = new IndexBuildManager(env);
+      } catch (IOException e) {
+          LOG.warn("failed to new index build manager", e);
+      }
+      //init index writer
+      // Clone the config since it is shared
+      Configuration clonedConfig = PropertiesUtil.cloneConfig(env.getConfiguration());
+      /*
+       * Set the rpc controller factory so that the HTables used by IndexWriter would
+       * set the correct priorities on the remote RPC calls.
+       */
+      clonedConfig.setClass(RpcControllerFactory.CUSTOM_CONTROLLER_CONF_KEY,
+          InterRegionServerIndexRpcControllerFactory.class, RpcControllerFactory.class);
+      // lower the number of rpc retries.  We inherit config from HConnectionManager#setServerSideHConnectionRetries,
+      // which by default uses a multiplier of 10.  That is too many retries for our synchronous index writes
+      clonedConfig.setInt(HConstants.HBASE_CLIENT_RETRIES_NUMBER,
+          env.getConfiguration().getInt(INDEX_WRITER_RPC_RETRIES_NUMBER,
+              DEFAULT_INDEX_WRITER_RPC_RETRIES_NUMBER));
+      clonedConfig.setInt(HConstants.HBASE_CLIENT_PAUSE, env.getConfiguration()
+          .getInt(INDEX_WRITER_RPC_PAUSE, DEFAULT_INDEX_WRITER_RPC_PAUSE));
+      DelegateRegionCoprocessorEnvironment indexWriterEnv = new DelegateRegionCoprocessorEnvironment(clonedConfig, env);
+      // setup the actual index writer
+      String serverName = env.getRegionServerServices().getServerName().getServerName();
+      try {
+          this.writer = new IndexWriter(indexWriterEnv, serverName + "-index-writer");
+      } catch (IOException e) {
+          LOG.warn("Failed to init index writer ", e);
```

Comment:
Again, why catch here ?

Reply:
> Again, why catch here ?

the postOpen() of hbase coprocessor can not bubble up this catch, only print log in this.